worker/src/DepLibUring.cpp

```diff
@@ -0,0 +1,439 @@
+{
+	MS_TRACE();
+
+	if (!DepLibUring::liburing)
+	{
+		return;
+	}
+
+	DepLibUring::liburing->StartPollingCQEs();
+}
+
+void DepLibUring::StopPollingCQEs()
+{
+	MS_TRACE();
+
+	if (!DepLibUring::liburing)
+	{
+		return;
+	}
+
+	DepLibUring::liburing->StopPollingCQEs();
+}
+
+bool DepLibUring::PrepareSend(
+  int sockfd, const void* data, size_t len, const struct sockaddr* addr, onSendCallback* cb)
+{
+	MS_TRACE();
+
+	MS_ASSERT(DepLibUring::liburing, "DepLibUring::liburing is not set");
+
+	return DepLibUring::liburing->PrepareSend(sockfd, data, len, addr, cb);
+}
+
+bool DepLibUring::PrepareWrite(
+  int sockfd, const void* data1, size_t len1, const void* data2, size_t len2, onSendCallback* cb)
+{
+	MS_TRACE();
+
+	MS_ASSERT(DepLibUring::liburing, "DepLibUring::liburing is not set");
+
+	return DepLibUring::liburing->PrepareWrite(sockfd, data1, len1, data2, len2, cb);
+}
+
+void DepLibUring::Submit()
+{
+	MS_TRACE();
+
+	if (!DepLibUring::liburing)
+	{
+		return;
+	}
+
+	DepLibUring::liburing->Submit();
+}
+
+void DepLibUring::SetActive()
+{
+	MS_TRACE();
+
+	if (!DepLibUring::liburing)
+	{
+		return;
+	}
+
+	DepLibUring::liburing->SetActive();
+}
+
+bool DepLibUring::IsActive()
+{
+	MS_TRACE();
+
+	if (!DepLibUring::liburing)
+	{
+		return false;
+	}
+
+	return DepLibUring::liburing->IsActive();
+}
+
+/* Instance methods. */
+
+DepLibUring::LibUring::LibUring()
+{
+	MS_TRACE();
+
+	/**
+	 * IORING_SETUP_SINGLE_ISSUER: A hint to the kernel that only a single task
+	 * (or thread) will submit requests, which is used for internal optimisations.
+	 */
+
+	unsigned int flags = IORING_SETUP_SINGLE_ISSUER;
+
+	// Initialize io_uring.
+	auto err = io_uring_queue_init(DepLibUring::QueueDepth, std::addressof(this->ring), flags);
+
+	if (err < 0)
+	{
+		MS_THROW_ERROR("io_uring_queue_init() failed: %s", std::strerror(-err));
+	}
+
+	// Create an eventfd instance.
+	this->efd = eventfd(0, 0);
+
+	if (this->efd < 0)
+	{
+		MS_THROW_ERROR("eventfd() failed: %s", std::strerror(-this->efd));
+	}
+
+	err = io_uring_register_eventfd(std::addressof(this->ring), this->efd);
+
+	if (err < 0)
+	{
+		MS_THROW_ERROR("io_uring_register_eventfd() failed: %s", std::strerror(-err));
+	}
+
+	// Initialize available UserData entries.
+	for (size_t i{ 0 }; i < DepLibUring::QueueDepth; ++i)
+	{
+		this->availableUserDataEntries.push(i);
+	}
+}
+
+DepLibUring::LibUring::~LibUring()
+{
+	MS_TRACE();
+
+	// Close the event file descriptor.
+	const auto err = close(this->efd);
+
+	if (err != 0)
+	{
+		MS_ABORT("close() failed: %s", std::strerror(-err));
+	}
+
+	// Close the ring.
+	io_uring_queue_exit(std::addressof(this->ring));
+}
+
+void DepLibUring::LibUring::Dump() const
+{
+	MS_TRACE();
+
+	MS_DUMP("<LibUring>");
+	MS_DUMP("  sqeProcessCount        : %" PRIu64, this->sqeProcessCount);
+	MS_DUMP("  sqeMissCount           : %" PRIu64, this->sqeMissCount);
+	MS_DUMP("  userDataMissCount      : %" PRIu64, this->sqeMissCount);
+	MS_DUMP("</LibUring>");
```

Comment:
```suggestion
	MS_DUMP("<DepLibUring>");
	MS_DUMP("  sqeProcessCount   : %" PRIu64, this->sqeProcessCount);
	MS_DUMP("  sqeMissCount      : %" PRIu64, this->sqeMissCount);
	MS_DUMP("  userDataMissCount : %" PRIu64, this->sqeMissCount);
	MS_DUMP("</DepLibUring>");
```

But why do we need this? This is not useful. I expected this info in the worker dump (meaning in the FBS dump response).

Reply:
It's just a mock method. The important thing is that it holds the values.

OK, I'll add this to the worker dump data.

Reply:
It's `LibUring` class where we are doing the `Dump()`.

Reply:
I'm removing this anyway.